spark/src/main/kotlin/com/adevinta/spark/components/text/TextLink.kt

```diff
@@ -0,0 +1,234 @@
+ * from [style] will be used instead.
+ *
+ * Additionally, for [color], if [color] is not set, and [style] does not have a color, then
+ * [LocalContentColor] will be used.
+ *
+ * @param textFull the full text to be displayed
+ * @param textLink the text to be underlined as link (it will be the clickable part
+ * WARNING if textFull does not include textLink it will @throws IndexOutOfBoundsException
+ *
+ * @param modifier the [Modifier] to be applied to this layout node
+ * @param colorText [Color] to apply to the text. If [Color.Unspecified], and [style] has no color set,
+ * this will be [LocalContentColor].
+ * @param colorLink [Color] to apply to the link. by default if no color specified it will
+ * be assigned to the value of @param colorText
+ * @param fontSize the size of glyphs to use when painting the text. See [TextStyle.fontSize].
+ * @param fontStyle the typeface variant to use when drawing the letters (e.g., italic).
+ * See [TextStyle.fontStyle].
+ * @param fontWeight the typeface thickness to use when painting the text (e.g., [FontWeight.Bold]).
+ * @param fontFamily the font family to be used when rendering the text. See [TextStyle.fontFamily].
+ * @param letterSpacing the amount of space to add between each letter.
+ * See [TextStyle.letterSpacing].
+ * @param textAlign the alignment of the text within the lines of the paragraph.
+ * See [TextStyle.textAlign].
+ * @param lineHeight line height for the [Paragraph] in [TextUnit] unit, e.g. SP or EM.
+ * See [TextStyle.lineHeight].
+ * @param overflow how visual overflow should be handled.
+ * @param softWrap whether the text should break at soft line breaks. If false, the glyphs in the
+ * text will be positioned as if there was unlimited horizontal space. If [softWrap] is false,
+ * [overflow] and TextAlign may have unexpected effects.
+ * @param maxLines an optional maximum number of lines for the text to span, wrapping if
+ * necessary. If the text exceeds the given number of lines, it will be truncated according to
+ * [overflow] and [softWrap]. If it is not null, then it must be greater than zero.
+ * @param minLines The minimum height in terms of minimum number of visible lines. It is required
+ * that 1 <= [minLines] <= [maxLines].
+ * @param onTextLayout callback that is executed when a new text layout is calculated. A
+ * [TextLayoutResult] object that callback provides contains paragraph information, size of the
+ * text, baselines and other details. The callback can be used to add additional decoration or
+ * functionality to the text. For example, to draw selection around the text.
+ * @param style style configuration for the text such as color, font, line height etc.
+ * @param onClick callback when textLink part is clicked.
+ */
+
+@Composable
+public fun TextLink(
+    textFull: String,
+    textLink: String,
+    modifier: Modifier = Modifier,
+    colorText: Color = Color.Unspecified,
+    colorLink: Color = colorText,
```

Comment:
The link color can also be defined with the string res annotations that we have in Spark that handles our tokens and would allow for a better customization without having for us to support the whole `Text` Api